br: CI fail due to test conflicts

The CI test `idc-jenkins-ci-tidb/check_dev_2 ` may fail when checking safepoint.
This is because there is gc safepoint registered by other concurrent test.
```
=== RUN   TestOperator

    operator_test.go:80: 

        	Error Trace:	tests/realtikvtest/brietest/operator_test.go:80

        	            				tests/realtikvtest/brietest/operator_test.go:195

        	Error:      	the service gc safepoint exists

        	Test:       	TestOperator

        	Messages:   	it is struct { ServiceID string "json:\"service_id\""; ExpiredAt int64 "json:\"expired_at\""; SafePoint int64 "json:\"safe_point\"" }{ServiceID:"backup-stream-TestPiTRAndBackupInSQL-1", ExpiredAt:1739795773, SafePoint:456075134474649603}

--- FAIL: TestOperator (0.00s)

=== RUN   TestFailure

    operator_test.go:80: 

        	Error Trace:	tests/realtikvtest/brietest/operator_test.go:80

        	            				tests/realtikvtest/brietest/operator_test.go:248

        	Error:      	the service gc safepoint exists

        	Test:       	TestFailure

        	Messages:   	it is struct { ServiceID string "json:\"service_id\""; ExpiredAt int64 "json:\"expired_at\""; SafePoint int64 "json:\"safe_point\"" }{ServiceID:"backup-stream-TestPiTRAndBackupInSQL-1", ExpiredAt:1739795773, SafePoint:456075134474649603}

--- FAIL: TestFailure (0.00s)
```

--- a/br/pkg/task/operator/config.go
+++ b/br/pkg/task/operator/config.go
@@ -32,9 +32,9 @@ const (
 type PauseGcConfig struct {
 	task.Config
 
-	SafePoint uint64        `json:"safepoint" yaml:"safepoint"`
-	SafePointID string      `json:"safepoint-id" yaml:"safepoint-id"`
-	TTL       time.Duration `json:"ttl" yaml:"ttl"`
+	SafePoint   uint64        `json:"safepoint" yaml:"safepoint"`
+	SafePointID string        `json:"safepoint-id" yaml:"safepoint-id"`
+	TTL         time.Duration `json:"ttl" yaml:"ttl"`
 
 	OnAllReady func() `json:"-" yaml:"-"`
 	OnExit     func() `json:"-" yaml:"-"`

--- a/br/pkg/task/operator/config.go
+++ b/br/pkg/task/operator/config.go
@@ -33,6 +33,8 @@ type PauseGcConfig struct {
 	task.Config
 
 	SafePoint   uint64        `json:"safepoint" yaml:"safepoint"`
+	// SafePointID is used to identify a specific safepoint.
+	// This field is only used in ***TEST*** now, you shouldn't use it in the src codes.
 	SafePointID string        `json:"safepoint-id" yaml:"safepoint-id"`
 	TTL         time.Duration `json:"ttl" yaml:"ttl"`
 

--- a/br/pkg/task/operator/config.go
+++ b/br/pkg/task/operator/config.go
@@ -33,6 +33,7 @@ type PauseGcConfig struct {
 	task.Config
 
 	SafePoint uint64        `json:"safepoint" yaml:"safepoint"`
+	SafePointID string      `json:"safepoint-id" yaml:"safepoint-id"`
 	TTL       time.Duration `json:"ttl" yaml:"ttl"`
 
 	OnAllReady func() `json:"-" yaml:"-"`

--- a/br/pkg/task/operator/config.go
+++ b/br/pkg/task/operator/config.go
@@ -32,7 +32,7 @@ const (
 type PauseGcConfig struct {
 	task.Config
 
-	SafePoint   uint64        `json:"safepoint" yaml:"safepoint"`
+	SafePoint uint64 `json:"safepoint" yaml:"safepoint"`
 	// SafePointID is used to identify a specific safepoint.
 	// This field is only used in ***TEST*** now, you shouldn't use it in the src codes.
 	SafePointID string        `json:"safepoint-id" yaml:"safepoint-id"`

--- a/br/pkg/task/operator/prepare_snap.go
+++ b/br/pkg/task/operator/prepare_snap.go
@@ -123,7 +123,7 @@ func hintAllReady() {
 
 // AdaptEnvForSnapshotBackup blocks the current goroutine and pause the GC safepoint and remove the scheduler by the config.
 // This function will block until the context being canceled.
-func AdaptEnvForSnapshotBackup(ctx context.Context, cfg *PauseGcConfig) error {
+func AdaptEnvForSnapshotBackup(ctx context.Context, cfg *PauseGcConfig, spID string) error {
 	utils.DumpGoroutineWhenExit.Store(true)
 	mgr, err := dialPD(ctx, &cfg.Config)
 	if err != nil {
@@ -153,7 +153,7 @@ func AdaptEnvForSnapshotBackup(ctx context.Context, cfg *PauseGcConfig) error {
 	defer cx.Close()
 
 	initChan := make(chan struct{})
-	cx.run(func() error { return pauseGCKeeper(cx) })
+	cx.run(func() error { return pauseGCKeeper(cx, spID) })
 	cx.run(func() error {
 		log.Info("Pause scheduler waiting all connections established.")
 		select {
@@ -217,10 +217,10 @@ func pauseAdminAndWaitApply(cx *AdaptEnvForSnapshotBackupContext, afterConnectio
 	return nil
 }
 
-func pauseGCKeeper(cx *AdaptEnvForSnapshotBackupContext) (err error) {
+func pauseGCKeeper(cx *AdaptEnvForSnapshotBackupContext, spID string) (err error) {
 	// Note: should we remove the service safepoint as soon as this exits?
 	sp := utils.BRServiceSafePoint{
-		ID:       utils.MakeSafePointID(),
+		ID:       spID,
 		TTL:      int64(cx.cfg.TTL.Seconds()),
 		BackupTS: cx.cfg.SafePoint,
 	}

--- a/br/pkg/task/operator/prepare_snap.go
+++ b/br/pkg/task/operator/prepare_snap.go
@@ -123,7 +123,7 @@ func hintAllReady() {
 
 // AdaptEnvForSnapshotBackup blocks the current goroutine and pause the GC safepoint and remove the scheduler by the config.
 // This function will block until the context being canceled.
-func AdaptEnvForSnapshotBackup(ctx context.Context, cfg *PauseGcConfig, spID string) error {
+func AdaptEnvForSnapshotBackup(ctx context.Context, cfg *PauseGcConfig) error {
 	utils.DumpGoroutineWhenExit.Store(true)
 	mgr, err := dialPD(ctx, &cfg.Config)
 	if err != nil {
@@ -153,7 +153,7 @@ func AdaptEnvForSnapshotBackup(ctx context.Context, cfg *PauseGcConfig, spID str
 	defer cx.Close()
 
 	initChan := make(chan struct{})
-	cx.run(func() error { return pauseGCKeeper(cx, spID) })
+	cx.run(func() error { return pauseGCKeeper(cx, cfg.SafePointID) })
 	cx.run(func() error {
 		log.Info("Pause scheduler waiting all connections established.")
 		select {

--- a/tests/realtikvtest/brietest/BUILD.bazel
+++ b/tests/realtikvtest/brietest/BUILD.bazel
@@ -21,6 +21,7 @@ go_test(
         "//br/pkg/summary",
         "//br/pkg/task",
         "//br/pkg/task/operator",
+        "//br/pkg/utils",
         "//pkg/config",
         "//pkg/domain",
         "//pkg/executor",

--- a/tests/realtikvtest/brietest/operator_test.go
+++ b/tests/realtikvtest/brietest/operator_test.go
@@ -247,8 +247,8 @@ func TestOperator(t *testing.T) {
 		}
 	}, 10*time.Second, time.Second)
 
+	verifySchedulerNotStopped(req, cfg)
 	verifyTargetGCSafePointNotExist(req, cfg, spID)
-	verifyTargetGCSafePointExist(req, cfg, spID)
 }
 
 func TestFailure(t *testing.T) {
@@ -278,6 +278,6 @@ func TestFailure(t *testing.T) {
 	err := operator.AdaptEnvForSnapshotBackup(ctx, &cfg, spID)
 	require.Error(t, err)
 
-	verifyTargetGCSafePointNotExist(req, cfg, spID)
 	verifySchedulerNotStopped(req, cfg)
+	verifyTargetGCSafePointNotExist(req, cfg, spID)
 }

--- a/tests/realtikvtest/brietest/operator_test.go
+++ b/tests/realtikvtest/brietest/operator_test.go
@@ -28,6 +28,7 @@ import (
 	"github.com/pingcap/kvproto/pkg/kvrpcpb"
 	"github.com/pingcap/tidb/br/pkg/task"
 	"github.com/pingcap/tidb/br/pkg/task/operator"
+	"github.com/pingcap/tidb/br/pkg/utils"
 	"github.com/stretchr/testify/require"
 	"github.com/tikv/client-go/v2/oracle"
 	pd "github.com/tikv/pd/client"
@@ -77,6 +78,27 @@ func verifyGCNotStopped(t *require.Assertions, cfg operator.PauseGcConfig) {
 	t.NoError(getJSON(pdAPI(cfg, serviceGCSafepointPrefix), &result))
 	for _, sp := range result.SPs {
 		if sp.ServiceID != "gc_worker" {
+			t.FailNowf("the service gc safepoint exists", "it is %#v", sp)
+		}
+	}
+}
+
+func verifyTargetGCSafePointExist(t *require.Assertions, cfg operator.PauseGcConfig, spID string) {
+	var result GcSafePoints
+	t.NoError(getJSON(pdAPI(cfg, serviceGCSafepointPrefix), &result))
+	for _, sp := range result.SPs {
+		if sp.ServiceID == spID {
+			return
+		}
+	}
+	t.FailNowf("the service gc safepoint does not exist", "it is %#v", result)
+}
+
+func verifyTargetGCSafePointNotExist(t *require.Assertions, cfg operator.PauseGcConfig, spID string) {
+	var result GcSafePoints
+	t.NoError(getJSON(pdAPI(cfg, serviceGCSafepointPrefix), &result))
+	for _, sp := range result.SPs {
+		if sp.ServiceID == spID {
 			t.FailNowf("the service gc safepoint exists", "it is %#v", sp)
 		}
 	}
@@ -192,13 +214,14 @@ func TestOperator(t *testing.T) {
 		},
 	}
 
-	verifyGCNotStopped(req, cfg)
+	spID := utils.MakeSafePointID()
+	verifyTargetGCSafePointNotExist(req, cfg, spID)
 	verifySchedulerNotStopped(req, cfg)
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	go func() {
-		req.NoError(operator.AdaptEnvForSnapshotBackup(ctx, &cfg))
+		req.NoError(operator.AdaptEnvForSnapshotBackup(ctx, &cfg, spID))
 	}()
 	req.Eventually(func() bool {
 		select {
@@ -210,6 +233,7 @@ func TestOperator(t *testing.T) {
 	}, 10*time.Second, time.Second)
 
 	verifyGCStopped(req, cfg)
+	verifyTargetGCSafePointExist(req, cfg, spID)
 	verifyLightningStopped(req, cfg)
 	verifySchedulersStopped(req, cfg)
 	cancel()
@@ -223,8 +247,8 @@ func TestOperator(t *testing.T) {
 		}
 	}, 10*time.Second, time.Second)
 
-	verifySchedulerNotStopped(req, cfg)
-	verifyGCNotStopped(req, cfg)
+	verifyTargetGCSafePointNotExist(req, cfg, spID)
+	verifyTargetGCSafePointExist(req, cfg, spID)
 }
 
 func TestFailure(t *testing.T) {
@@ -245,14 +269,15 @@ func TestFailure(t *testing.T) {
 		SafePoint: oracle.GoTimeToTS(time.Now()),
 	}
 
-	verifyGCNotStopped(req, cfg)
+	spID := utils.MakeSafePointID()
+	verifyTargetGCSafePointNotExist(req, cfg, spID)
 	verifySchedulerNotStopped(req, cfg)
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	err := operator.AdaptEnvForSnapshotBackup(ctx, &cfg)
+	err := operator.AdaptEnvForSnapshotBackup(ctx, &cfg, spID)
 	require.Error(t, err)
 
-	verifyGCNotStopped(req, cfg)
+	verifyTargetGCSafePointNotExist(req, cfg, spID)
 	verifySchedulerNotStopped(req, cfg)
 }

--- a/tests/realtikvtest/brietest/operator_test.go
+++ b/tests/realtikvtest/brietest/operator_test.go
@@ -204,8 +204,8 @@ func TestOperator(t *testing.T) {
 		Config: task.Config{
 			PD: []string{"127.0.0.1:2379"},
 		},
-		TTL:       5 * time.Minute,
-		SafePoint: oracle.GoTimeToTS(time.Now()),
+		TTL:         5 * time.Minute,
+		SafePoint:   oracle.GoTimeToTS(time.Now()),
 		SafePointID: utils.MakeSafePointID(),
 		OnAllReady: func() {
 			close(rd)
@@ -265,8 +265,8 @@ func TestFailure(t *testing.T) {
 		Config: task.Config{
 			PD: []string{"127.0.0.1:2379"},
 		},
-		TTL:       5 * time.Minute,
-		SafePoint: oracle.GoTimeToTS(time.Now()),
+		TTL:         5 * time.Minute,
+		SafePoint:   oracle.GoTimeToTS(time.Now()),
 		SafePointID: utils.MakeSafePointID(),
 	}
 

--- a/tests/realtikvtest/brietest/operator_test.go
+++ b/tests/realtikvtest/brietest/operator_test.go
@@ -83,22 +83,22 @@ func verifyGCNotStopped(t *require.Assertions, cfg operator.PauseGcConfig) {
 	}
 }
 
-func verifyTargetGCSafePointExist(t *require.Assertions, cfg operator.PauseGcConfig, spID string) {
+func verifyTargetGCSafePointExist(t *require.Assertions, cfg operator.PauseGcConfig) {
 	var result GcSafePoints
 	t.NoError(getJSON(pdAPI(cfg, serviceGCSafepointPrefix), &result))
 	for _, sp := range result.SPs {
-		if sp.ServiceID == spID {
+		if sp.ServiceID == cfg.SafePointID {
 			return
 		}
 	}
 	t.FailNowf("the service gc safepoint does not exist", "it is %#v", result)
 }
 
-func verifyTargetGCSafePointNotExist(t *require.Assertions, cfg operator.PauseGcConfig, spID string) {
+func verifyTargetGCSafePointNotExist(t *require.Assertions, cfg operator.PauseGcConfig) {
 	var result GcSafePoints
 	t.NoError(getJSON(pdAPI(cfg, serviceGCSafepointPrefix), &result))
 	for _, sp := range result.SPs {
-		if sp.ServiceID == spID {
+		if sp.ServiceID == cfg.SafePointID {
 			t.FailNowf("the service gc safepoint exists", "it is %#v", sp)
 		}
 	}
@@ -206,6 +206,7 @@ func TestOperator(t *testing.T) {
 		},
 		TTL:       5 * time.Minute,
 		SafePoint: oracle.GoTimeToTS(time.Now()),
+		SafePointID: utils.MakeSafePointID(),
 		OnAllReady: func() {
 			close(rd)
 		},
@@ -214,14 +215,13 @@ func TestOperator(t *testing.T) {
 		},
 	}
 
-	spID := utils.MakeSafePointID()
-	verifyTargetGCSafePointNotExist(req, cfg, spID)
+	verifyTargetGCSafePointNotExist(req, cfg)
 	verifySchedulerNotStopped(req, cfg)
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	go func() {
-		req.NoError(operator.AdaptEnvForSnapshotBackup(ctx, &cfg, spID))
+		req.NoError(operator.AdaptEnvForSnapshotBackup(ctx, &cfg))
 	}()
 	req.Eventually(func() bool {
 		select {
@@ -233,7 +233,7 @@ func TestOperator(t *testing.T) {
 	}, 10*time.Second, time.Second)
 
 	verifyGCStopped(req, cfg)
-	verifyTargetGCSafePointExist(req, cfg, spID)
+	verifyTargetGCSafePointExist(req, cfg)
 	verifyLightningStopped(req, cfg)
 	verifySchedulersStopped(req, cfg)
 	cancel()
@@ -248,7 +248,7 @@ func TestOperator(t *testing.T) {
 	}, 10*time.Second, time.Second)
 
 	verifySchedulerNotStopped(req, cfg)
-	verifyTargetGCSafePointNotExist(req, cfg, spID)
+	verifyTargetGCSafePointNotExist(req, cfg)
 }
 
 func TestFailure(t *testing.T) {
@@ -267,17 +267,17 @@ func TestFailure(t *testing.T) {
 		},
 		TTL:       5 * time.Minute,
 		SafePoint: oracle.GoTimeToTS(time.Now()),
+		SafePointID: utils.MakeSafePointID(),
 	}
 
-	spID := utils.MakeSafePointID()
-	verifyTargetGCSafePointNotExist(req, cfg, spID)
+	verifyTargetGCSafePointNotExist(req, cfg)
 	verifySchedulerNotStopped(req, cfg)
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	err := operator.AdaptEnvForSnapshotBackup(ctx, &cfg, spID)
+	err := operator.AdaptEnvForSnapshotBackup(ctx, &cfg)
 	require.Error(t, err)
 
 	verifySchedulerNotStopped(req, cfg)
-	verifyTargetGCSafePointNotExist(req, cfg, spID)
+	verifyTargetGCSafePointNotExist(req, cfg)
 }

